core/src/test/java/org/elasticsearch/common/geo/GeoWKTShapeParserTests.java

```diff
@@ -0,0 +1,182 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.common.geo;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.LinearRing;
+import com.vividsolutions.jts.geom.MultiLineString;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.geom.Polygon;
+import org.apache.lucene.geo.GeoTestUtil;
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.geo.builders.EnvelopeBuilder;
+import org.elasticsearch.common.geo.builders.LineStringBuilder;
+import org.elasticsearch.common.geo.builders.MultiLineStringBuilder;
+import org.elasticsearch.common.geo.builders.MultiPointBuilder;
+import org.elasticsearch.common.geo.builders.MultiPolygonBuilder;
+import org.elasticsearch.common.geo.builders.PointBuilder;
+import org.elasticsearch.common.geo.builders.PolygonBuilder;
+import org.elasticsearch.common.geo.builders.ShapeBuilder;
+import org.elasticsearch.common.geo.parsers.GeoWKTParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.test.geo.RandomShapeGenerator;
+import org.locationtech.spatial4j.shape.Rectangle;
+import org.locationtech.spatial4j.shape.Shape;
+import org.locationtech.spatial4j.shape.ShapeCollection;
+import org.locationtech.spatial4j.shape.jts.JtsPoint;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.common.geo.builders.ShapeBuilder.SPATIAL_CONTEXT;
+
+/**
+ * Tests for {@code GeoWKTShapeParser}
+ */
+public class GeoWKTShapeParserTests extends BaseGeoParsingTestCase {
+
+    private static XContentBuilder toWKTContent(ShapeBuilder builder, boolean generateMalformed)
+            throws IOException {
+        String wkt = builder.toWKT();
+        if (generateMalformed) {
+            // malformed - extra paren
+            // TODO generate more malformed WKT
+            wkt += GeoWKTParser.RPAREN;
+        }
+        if (randomBoolean()) {
+            // test comments
+            wkt = "# " + wkt + "\n" + wkt;
+        }
+        return XContentFactory.jsonBuilder().value(wkt);
+    }
+
+    private void assertExpected(Shape expected, ShapeBuilder builder) throws IOException {
+        boolean isMalformed = randomBoolean();
```

Comment:
I would prefer at least one good and one malformed case per test run. Otherwise we only catch bigger errors on multiple repetions and not e.g. on PR tests.

Reply:
I'll add an explicit malformed test for common typos.

Reply:
:+1:

Reply:
thanks